Arc/Sage/TaskListScheduleManager.swift

```diff
@@ -113,25 +113,49 @@ public class TaskListScheduleManager {
         
         let reportIds: [RSDIdentifier] = [.availability, .testSchedule]
         var successCtr = reportIds.count
+        var completedCalled = false
         
         reportIds.forEach { (identifier) in
             self.getSingletonReport(reportId: identifier) { (report, error) in
                 if (error != nil) {
-                    completed(nil, nil, error)
+                    if (!completedCalled) {
+                        completed(nil, nil, error)
+                    }
+                    completedCalled = true
+                    return
                 }
                 
+                var errorStr: String? = nil
                 switch(identifier) {
                 case .availability:
-                    availabilityData = self.createWakeSleepScheduleRequestData(mostRecentReport: report)
+                    let result = self.createWakeSleepScheduleRequestData(mostRecentReport: report)
+                    availabilityData = result.0
+                    errorStr = result.1
                 case .testSchedule:
-                    testScheduleData = self.createTestScheduleRequestData(mostRecentReport: report)
+                    let result = self.createTestScheduleRequestData(mostRecentReport: report)
+                    testScheduleData = result.0
+                    errorStr = result.1
                 default:
                     debugPrint("Report id \(identifier) not supported in loadHistoryFromBridge")
                 }
                 
+                if let errorStrUnwrapped = errorStr {
+                    if (!completedCalled) {
+                        completed(nil, nil, errorStrUnwrapped)
+                    }
+                    completedCalled = true
+                    return
+                }
+                
                 successCtr -= 1
-                if (successCtr <= 0) {  // Check for done state
+                
+                // Check for done state
+                if (successCtr <= 0 && !completedCalled) {
```

Comment:
Why is this inside the foreach instead of after?

Reply:
The call to getSingletonReport inside the foreach is an asynchronous call, so if you put the check outside the foreach it would be called immediately and none of the asynchronous calls will have returned yet.